### PR TITLE
Split disko type into multiple files

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: types/default.nix [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+      nodev = lib.mkOption {
+        type = lib.types.attrsOf subTypes.nodev;
+        default = { };
+        description = "A non-block device";
+      };
+    };
+  };
+
+  subTypes = {
+    nodev = diskoLib.mkSubType [ ./nodev.nix ];
```
Comment (Lassulus) [inline types/default.nix:328]: why does this take a list?

Comment (Mic92) [reply types/default.nix:328]: modules

Comment (Mic92) [reply types/default.nix:327]: https://github.com/nix-community/disko/pull/131/files#diff-63b2750ca88f0bb1b20a657859ddd5df357da7d87777bfd4054db88de1a6a598R148

Comment (Lassulus) [reply types/default.nix:324]: but we don't need the list interface? it could be just a file?

Comment (Mic92) [reply types/default.nix]: Done.

**Review by Lassulus — APPROVED** [review]

**Comment by Lassulus** [comment]
bors merge

**Pull request merged**